services/monitor/monitor.sh

```
@@ -0,0 +1,368 @@
+save_container_logs() {
+    if ! container_exists; then
+        return 0
+    fi
+
+    # Ensure log directory exists
+    mkdir -p "$LOG_ARCHIVE_DIR" || {
+        error "Failed to create log archive directory: $LOG_ARCHIVE_DIR"
+        return 1
+    }
+
+    local timestamp=$(date '+%Y%m%d-%H%M%S')
```

Shellcheck (0.11.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)